### PR TITLE
add support for map type in Trino handler

### PR DESCRIPTION
**Pull request by satish-mittal**
## Description
Please include a summary of the change, motivation and context.

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: helm/values.yaml [hunk 1]
```diff
@@ -81,7 +81,7 @@ queryServiceConfig:
     postgresqlPasswordSecretKey: password
     postgresMaxConnectionAttempts: 200
     postgresConnectionRetryBackoff: 5s
-    trinogresConnectionString: jdbc:trino://localhost:8080/iceberg/iceberg_gcs
+    trinoConnectionString: jdbc:trino://localhost:8080/iceberg/iceberg_gcs
```
Comment (kotharironak) [inline helm/values.yaml:84]: So, the typo was only in values, right? The key in template is still `trinoConnectionString`

File: query-service-impl/src/main/java/org/hypertrace/core/query/service/trino/converters/DefaultColumnRequestConverter.java [hunk 2]
```diff
@@ -31,7 +31,7 @@
 
 /** Converts {@link QueryRequest} to Trino SQL query */
 class DefaultColumnRequestConverter implements ColumnRequestConverter {
-
+  private static final String MAP_KEY_OPERATOR = "element_at";
```
Comment (kotharironak) [inline query-service-impl/src/main/java/org/hypertrace/core/query/service/trino/converters/DefaultColumnRequestConverter.java:34]: nit: `element_at` is map function in trino world. Should we rename this to MAP_FUNCTION = ""element_at""

File: query-service-impl/src/main/java/org/hypertrace/core/query/service/trino/converters/DefaultColumnRequestConverter.java [hunk 3]
```diff
@@ -111,26 +111,35 @@ private String convertFilterToString(
               convertExpressionToString(filter.getRhs(), paramsBuilder, trinoExecutionContext));
           break;
         case CONTAINS_KEY:
+          builder.append(operator);
```
Comment (kotharironak) [inline query-service-impl/src/main/java/org/hypertrace/core/query/service/trino/converters/DefaultColumnRequestConverter.java:114]:
```suggestion
          builder.append(MAP_FUNCTION);
```

Comment (kotharironak) [reply query-service-impl/src/main/java/org/hypertrace/core/query/service/trino/converters/DefaultColumnRequestConverter.java:114]: if we treat `element_at` as function.

File: query-service-impl/src/main/java/org/hypertrace/core/query/service/trino/converters/DefaultColumnRequestConverter.java (hunk 2 above)
Comment (satish-mittal) [reply query-service-impl/src/main/java/org/hypertrace/core/query/service/trino/converters/DefaultColumnRequestConverter.java]: done

File: helm/values.yaml (hunk 1 above)
Comment (satish-mittal) [reply helm/values.yaml:84]: yes

File: query-service-impl/src/main/java/org/hypertrace/core/query/service/trino/converters/DefaultColumnRequestConverter.java (hunk 3 above)
Comment (satish-mittal) [reply query-service-impl/src/main/java/org/hypertrace/core/query/service/trino/converters/DefaultColumnRequestConverter.java:114]: `operator` is initialised by `convertOperatorToString()` in a generic way, which will return `MAP_FUNCTION` in this case

**Review by kotharironak — APPROVED** [review]

**Pull request merged**